pkg/network/setup/netpod/state.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package netpod
+
+import (
+	"fmt"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	neterrors "kubevirt.io/kubevirt/pkg/network/errors"
+)
+
+type stateCacheRUD interface {
```

Comment:
This comes from the well known `CRUD` [1] definition.
The only alternative which makes sense here would be `stateCacheReadWriteDelete`.
I could change it to this new name, but it is long and I am not sure it is worth the effort.

As the interface is local, limited to this file, it can be changed easily if someone will feel it is annoying.
If I'll have to send a change to the PR, I won't object to change it.

[1] https://en.wikipedia.org/wiki/Create,_read,_update_and_delete

Reply:
Thanks for the reference, I am familiar with CRUD definition.

Having RUD only doesn't ring a bell as CRUD which is well know definition.
The long name form has the same problems as the short one.

You can leave the name `stateCache` or `stateCacher` (if you want to be strict about idiomatic go interface naming..).
I suggest it in this PR because its a good opportunity to clear this up, and there are more changes then just move things around, but yeah.. if you really don't what to do it now I can do it later.

Reply:
> You can leave the name `stateCache` or `stateCacher`

Interfaces should express behavior and not "er" the original name.
We have in the project a pattern of "intefacing" whole objects, which is wrong.

I realized this PR will need to pass another test round by the CI because many things got in already, so it will rebase it on main and re-run the tests. So I will change it.

Done

Reply:
> Interfaces should express behavior and not "er" the original name.

This is why I mentioned `stateCacher`.
The log name form suffer from the same issue as the one with initial.

> We have in the project a pattern of "intefacing" whole objects, which is wrong

I guess there are cases it fits, I recall we saw this pattern in other projects as well.

Reply:
Behavior means it expresses what the interface exposes and not to which object it belongs to. The whole point of an interface is to decouple it from the concrete object.

The fact you saw things on other projects is not a sign it is correct.
I see more miss-usage of interfaces over good usage. I suggest each incident to be well reason for, that should help in avoiding bad usage copy.